internal/adapter/lsp/server.go

```diff
@@ -389,6 +395,75 @@ func NewServer(opts ServerOpts) *Server {
 		return actions, nil
 	}
 
+	handler.TextDocumentReferences = func(context *glsp.Context, params *protocol.ReferenceParams) ([]protocol.Location, error) {
+		doc, ok := server.documents.Get(params.TextDocument.URI)
+		if !ok {
+			return nil, nil
+		}
+
+		link, err := doc.DocumentLinkAt(params.Position)
+		if link == nil || err != nil {
+			return nil, err
+		}
+
+		notebook, err := server.notebookOf(doc)
+		if err != nil {
+			return nil, err
+		}
+
+		target, err := server.noteForHref(link.Href, doc, notebook)
+		if link == nil || target == nil || err != nil {
+			return nil, err
+		}
+
+		opts := core.NoteFindOpts{}
+		p, err := notebook.RelPath(target.Path)
+		if err != nil {
+			return nil, err
+		}
+
+		opts.LinkTo = &core.LinkFilter{
+			Paths:       []string{p},
+			Negate:      false,
+			Recursive:   true,
+			MaxDistance: 1,
+		}
```

Comment:
`Recursive` can be used to look for indirect links across several notes. But since you used a `MaxDistance` of 1, it's like having recursive disabled, which is the default. It's a bit more efficient with SQLite to disable recursive when not needed.

```suggestion
		p, err := notebook.RelPath(target.Path)
		if err != nil {
			return nil, err
		}

		opts := core.NoteFindOpts{
			LinkTo: &core.LinkFilter{Paths: []string{p}},
		}
```